content/post/against-ab-tests.md

```diff
@@ -0,0 +1,93 @@
+
+### Software Tools
+
+In this brave new world, where we serve different content to different user groups (in proportions that may shift over time) we will need tools for both building and understanding our software. 
+The most obvious immediate impact of such a paradigm would be a dramatic increase in the code surface-area of a project. Instead of retiring old code paths after a test has been run, we actually have to keep them running (potentially forever). This is scary!  
+What we need are ways to build applications _more modularly_ so that we can develop, test, deploy, and maintain new code-paths (e.g., new test arms) perpetually.
+
+Because we want to be able to fork users down different code paths based on their attributes (with a potentially large number of forks for a given customer journey) we need to engineer an architecture that can support this forking. We need a _centralized decision engine_ that can decide which path a given user should be put on, and we need the path components to be sufficiently interchangeable so that users have a seamless journey even when the path components were developed independently without a view of a holistic "user journey".
+
+Finally, without a single, holistic, user journey, we need tools for product managers and designers to be able to envision the customer journey in a garden of forking paths. How do we ideate and size new features? How do we trace what steps a given user even experienced when they used our application? How do we prevent the application from becoming an opaque mess of code spaghetti?
+
+### Communication and Education
+
+This new way of thinking about software development will be especially challenging for non-engineers to conceptualize and think about. Executives have been trained to think about a single user journey with a single brand voice and a single, uniform experience. As we start personalizing the user experience, the ability to reason about the application holistically will go away.
+
+We will need to educate our stakeholders on the value of this approach, and we will have to help them reason about the customer journey and the brand voice in such a context. We will need to develop methods of explaining which journeys are most common, and give them the tools to explore the application while "assuming the identity" of a particular subgroup so that they can experience the personalized application from the perspective of different customers.
+
+### Statistical Tools
+
+Perhaps most obviously, in a world without A/B tests, we will have to throw out many of the tools we've traditionally used for optimizing our web applications. All of the work we have done to educate product managers and marketers on the appropriate way to run and interpret A/B tests will be like dust in the wind.
```

Comment:
Are you suggesting the removal of testing in general or just the old 2-system, winner takes all paradigm? In this case testing can mean either classic hypothesis testing or testing insofar as being able to make decisions in a structured way.

Reply:
This paragraph was targeted at classical hypothesis testing. The audience I'm thinking of is someone who is like "jeez, we _just_ got our PMs on board with testing at all, now we have to do a whole new thing?"

Does that make sense? Would love a suggestion on how to make that more clear